modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1152 @@
+    toolbar: "forecolor backcolor tableprops",
+    color_cols: 7, // <-- option was not respected
+});
+----
+
+In {productname} 6.4.1, the color swatches opened by the `forecolor` and `backcolor` toolbar buttons have the amount of columns set by the `color_cols` option, if it is set.
+
+=== Drag and dropping the last noneditable element out of its parent block would not properly pad the parent block element
+//#TINY-9606
+
+In previous versions of {productname}, dragging the last child element with a `contentedtiable="false"` attribute out of a block element would prevent the user from placing the caret back into that source block.
+
+As a fix, {productname} 6.4.1, added a new condition which validates if the source element is empty when the drop operation occurs. If it is empty then it will be padded with a `BR` element so that it's possible to place the caret inside the block. 
+
+In {productname} 6.4.1, dragging and dropping elements that are `contenteditable="false"` out of they're parent blocks now correctly pad the parent block element.
+
+=== Nonbreaking space do not get highlighted when enabling show invisible characters
+//#TINY-9685
+
+In previous versions of {productname}, it was discovered that nonbreaking spaces were not highlighting correctly when the "show invisible characters" plugin was enabled.
+
+This issue caused not only confusion with the intended text layout, but impacted the overall user experience. To resolve this issue, {productname} introduced a fix to ensure that all invisible characters, including nonbreaking spaces, are accurately displayed when the plugin is enabled.
+
+=== Non breaking space is added after backspace below text pattern heading
+//#TINY-9603
+
+A recently discovered issue was identified in previous versions of {productname}, that affected the text formatting feature for **Text Pattern**. When a user would apply a text pattern heading such as `# Title`, followed by pressing "enter", the issue would the occur when the backspace key was pressed. As a result, a non-breaking space was being inserted to the beginning of the line instead. This unintended addition of a Non-breaking space caused formatting inconsistencies and errors.
+
+In {productname} 6.4.1, pressing backspace after adding a Text Pattern, now formats the content as expected.
+
+=== Incorrect event target for dragend and drop events when dragging and dropping noneditable elements within editor
+//#TINY-9599
+
+A bug was discovered in previous version of {productname}, that would arise when dragging and dropping `contenteditable="false"` elements within the editor was that the target of the dragstart, dragend or drop events would be the editor instance itself, rather than the expected DOM target.
+
+In {productname} 6.4.1, the events are now simulated based on the mouse event that produced them, meaning that the target and other properties, such as mouse coordinates, are now properly produced for the simulated drag and drop events.
```

Comment:
```suggestion
In previous versions of {productname}, when dragging and dropping `contenteditable="false"` elements within the editor, the target of the `dragstart`, `dragend`, and `drop` events would be the editor instance itself, rather than the expected DOM target.

In {productname} 6.4.1, the events are now simulated based on the mouse event that produced them, meaning that the target and other properties, such as mouse coordinates, are now properly produced for the simulated `dragstart`, `dragend`, and `drop` events.
```